aiven/resource_vpc_peering_connection.go

```diff
@@ -159,19 +160,65 @@ func resourceVPCPeeringConnectionCreate(d *schema.ResourceData, m interface{}) e
 	}
 
 	timeout := d.Timeout(schema.TimeoutCreate)
-	res, err := w.Conf(timeout).WaitForState()
+	res, err := w.Conf(timeout).WaitForStateContext(ctx)
 	if err != nil && !aiven.IsAlreadyExists(err) {
-		return err
+		return diag.Errorf("Error waiting for VPC peering connection creation: %s", err)
 	}
 
 	pc = res.(*aiven.VPCPeeringConnection)
+	if pc.State != "ACTIVE" {
+		switch pc.State {
+		case "PENDING_PEER":
+			return diag.Errorf("Aiven platform has created a connection to the specified "+
+				"peer successfully in the cloud, but the connection is not active until the user "+
+				"completes the setup in their cloud account. The steps needed in the user cloud "+
+				"account depend on the used cloud provider. Find more in the state info: %s",
+				stateInfoToString(pc.StateInfo))
```

Comment:
I wonder about this pending peer handling as this is the typical state the connection ends up in. Maybe check with @mwfrojdman what the expected correct flows would be. (Maybe it is this but I'm not sure.)

Reply:
The happy path is
- create vpc peering connection through the aiven api (initial state `APPROVED`)
- wait for aiven to build the connection (-> state `PENDING_PEER`)
- Perform cloud provider specific actions in your own account to complete the setup
- Wait for the Aiven peering connection to enter state `ACTIVE`

On Azure some additional setup is needed before the steps described above

If the peering connection enters the state `INVALID_SPECIFICATION`, the peer vpc probably does doesn't exist, or some permissions are lacking depending on cloud provider. The error message may give some hints

Reply:
The idea behind this code is "block" the creation of ``` resource until it is in the `ACTIVE` state. The motivation was to notify a user whenever he/she need to perform some cloud provider-specific actions or change something in the configuration. 

Imaginable path: 
- When a user is creating `aiven_vpc_peering_connection` provider waits until resource reaches any state other than `APPROVED`.
- Let's say after `APPROVED` state we've got `PENDING_PEER`, in this case, creation of `aiven_vpc_peering_connection` will fail and a user will get a message with an invitation to perform some cloud provider-specific actions, also we print contents of the `state_info`. 
- A user performs required cloud provider-specific actions and runs `terraform apply` again.
- `aiven_vpc_peering_connection` is in the `ACTIVE` state, everyone is happy.

Reply:
@rikonen @mwfrojdman what would be desired behaviour for `PENDING_PEER` in Terraform Provider taking into consideration this thread?

Reply:
The AWS provider has an [accepter resource](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/vpc_peering_connection_accepter). I'm not sure how it works, but ideally the user could do something like:
1. Create the vpc peering connection (pcx) (like now)
2. When the pcx goes into PENDING_PEER on AWS, the user could use the AWS accepter resource to accept the pcx request in their account
3. The Aiven pcx resource (or is another resource needed? my TF skills are not that great) would depend on on the accepter resource and this resource would be sure to have state active

Pseudo example of how to use the accepter (not sure the `my_aiven_pcx` fields are very correct)
```
resource "aws_vpc_peering_connection_accepter" "my_accepter" {
  vpc_peering_connection_id = my_aiven_pcx.aws_vpc_peering_connection_id
  auto_accept               = true
}
```